File: week2/csharp/test/ProductTest.cs
using csharp.model;

namespace test
{
    [TestClass]
    public class ProductTest
    {
        [TestMethod]
        public void WhenCallingDefaultConstructor_ShouldCreateEmptyProduct()
        {
            Product product = new();

            Assert.AreEqual("", product.BarCode);
            Assert.AreEqual("", product.Description);
            Assert.AreEqual(0.00, product.Price);
            Assert.AreEqual(0, product.Count);
        }

        [TestMethod]
        public void WhenCallingParamsConstructor_ShouldCreateProductWithSpecifiedParams()
        {
            Product product = new()
            {
                BarCode = "1234",
                Description = "Description",
                Price = 2.56,
                Count = 5
            };

            Assert.AreEqual("1234", product.BarCode);
            Assert.AreEqual("Description", product.Description);
            Assert.AreEqual(2.56, product.Price);
            Assert.AreEqual(5, product.Count);
        }
    }
}

File: week2/ts/test/product.test.ts
import { describe, it, expect } from "vitest";
import { Product } from "../src/model/product.model";

describe("Product", () => {
  describe("when calling default `constructor`", () => {
    it("should create an empty product", () => {
      const product = new Product();

      expect(product.getId()).toBe("");
      expect(product.description).toBe("");
      expect(product.price).toBe(0.0);
      expect(product.count).toBe(0);
    });
  });

  describe("when calling `constructor` with params", () => {
    it("should create a product with specified properties", () => {
        const product = new Product("1234","Description",2.56,5);

        expect(product.getId()).toBe("1234");
      expect(product.description).toBe("Description");
      expect(product.price).toBe(2.56);
      expect(product.count).toBe(5);
    })
  })
});
